docs/cli-usage.md

```diff
@@ -0,0 +1,96 @@
+```sh
+kubectl run <pod-name> --image=nginx [--replicas=N] [--export [--external]]
+```
+
+This translates into a call to create a replication controller and optionally, a call to create a service which exposes the pods created by that replication controller.
+
+
+#### describe
+```sh
+kubectl describe <name>
+```
+
+Describe attempts a get against the three resource types (Pods, Replication Controllers, Services) and returns matching resource(s) for the named resource.  It ignores 404 errors, unless all of the get requests return 404.
+
+
+#### resize
+```sh
+kubectl resize <name> --replicas=N
+```
+
+Update is used to either update the size of a replication controller.
+
+#### export
+```sh
+kubectl export <service-name> <name> [--external] [--public-ip=<ip>] [--create-balancer=<bool>]
+```
+
+Export is used to create a service which represents a replicated set of pods.  First the replication controller is looked up using <name> then a new service called <service-name> is created.  If --external is specified, the service is also exposed on a public IP address, if --public-ip is specified, it is used as the ip address.  If --create-balancer is true (the default) an attempt to create a cloud load balancer with that IP address is performed.
+
+#### label
+```sh
+kubectl label <name> <key-1>=<value-1> ... <key-n>=<value-n>
+kubectl rmlabel <name> <key-1 ... <key-n>
+```
+
+Add, removes or updates labels to the api object named <name>. If more than one such object exists, returns an error, in which case the user must specify a complete path (e.g. ```pods/<name>```)
+
+
+#### delete
+```sh
+kubectl delete <name>
+```
+
+Deletes an api object specified by ```<name>```.  Prompts the user for confirmation before deleting.  
+
+
+### Usage Example
+Consider setting up the [Guestbook example](examples/guestbook/README.md).
+
+#### Create and export the master
+```sh
+kubectl run redis-master --image=dockerfile/redis --export
+```
+
+#### Create and export the slaves
+```sh
+kubectl run redis-slave --image=brendanburns/redis-slave --replicas=2 --export
+```
+
+#### Create and export the frontend
+```sh
+kubectl run frontend --image=php/redis --replicas=3 --export --external
+```
+
+### Resource Naming and Discovery
+
+In many places we accept a ```<name>``` for a resource.  Sometimes the type of resource is implicit in the command (e.g. ```run``` or ```export```) but often it is ambiguous. When it is ambiguous (e.g. ```delete```) the kubectl tool does most of the work of determining which resource the user is requesting.  
+
+Given this goal, the resource discovery algorithm works as follows:
+
+If ```<name>`` has no slashes '/' in it, it is both without namespace and without a resource type.  In this case we first do a search in each resource type (pods, replica controllers, services)and the default namespace ('default') for an object with that name.  If this search fails, we return "not found" and ask the user if the object exists in a namespace.
```

Comment:
I agree that the namespace defaulting should be handled by the existing context mechanism.

I agree we should support uids and label selectors everywhere as well as names.

I also agree that all our commands should accept multiple objects (#3050). For example:
```
kubectl get pods foo bar baz
kubectl delete pods foo bar baz
```

Where types and/or namespaces are specified on the command line, we should support URL suffixes and not design a new syntax. For example:
```
kubectl describe replicationcontrollers/foo services/foo-service
```

As for type discovery: I could easily imagine it being useful to do:
```
kubectl describe all
```
To describe all resources in the current namespace. We need to support at least resource listing (#2057) for that, and ideally API plugins (#991).

If we made it possible for LIST to return just names and not full resources (#2740), type discovery would be much more efficient.

Note that we also want to be able to drive commands (all commands, not just create/update) from configuration files, so that's another way to specify multiple objects.